addon/mixins/time-series-labeler.js

```diff
@@ -178,7 +171,49 @@ export default Ember.Mixin.create({
 
   //  This is the set of ticks on which labels appear.
   labelledTicks: Ember.computed('unfilteredLabelledTicks', 'tickFilter', function() {
-    return this.get('unfilteredLabelledTicks').filter(this.get('tickFilter'));
+    var ticks = this.get('unfilteredLabelledTicks');
+    return ticks.filter(this.get('tickFilter'));
+  }),
+
+  // We need a method to figure out the interval specifity
+  intervalSpecificity: Ember.computed('times', 'minTimeSpecificity', function(){
+    var ind1, ind2, domainTypes, maxNumberOfLabels, i, len, timeBetween;
+
+    // Now the real trick is if there is any allowance for minor ticks we should
+    // consider inflating the max allowed ticks to see if we can fit in a more
+    // specific domain.  Previous versions would increase the specifity one step
+    // which would then be cut out in filtering.
+    // A single minor tick alows us to double our capacity - 2 to triple
+    maxNumberOfLabels = this.get('maxNumberOfLabels') * (this.get('maxNumberOfMinorTicks') + 1);
+
+    // Find the segments that we'll test for (inclusive)
+    ind1 = this.get('DOMAIN_ORDERING').indexOf(this.get('minTimeSpecificity'));
+    ind2 = this.get('DOMAIN_ORDERING').indexOf(this.get('maxTimeSpecificity')) + 1;
+    // Refers to the metrics used for the labelling
+    domainTypes = ind2 < 0 ? this.get('DOMAIN_ORDERING').slice(ind1) : this.get('DOMAIN_ORDERING').slice(ind1, ind2);
```

Comment:
This line is pretty long, and I don't like the `?:` operator anyway because it often hides the underlying if/else. Can we change it to `if() {} else {}`?